---
title: "Group08 - 05_analysis_2" 
subtitle: "Volcano plot" 
author: "Julija" 
format: 
  html:
    output-dir: results
editor: visual
---

## Load packages

```{r}
library(tidyverse) 
library(ggplot2) 
library(ggrepel)
```

## Load data

```{r}
#loading data
expr_with_meta <- readr::read_tsv("../_raw/data/03_dat_aug.tsv")
```

```{r}
# as the averages are calculated for the replicates and the average values are shown for each replicate, here replicate 2 is removed as it has most of the same information and rpelicate 1
expr_df <- expr_with_meta |>
  filter(replicate == "Rep1")
```

## Selecting for shoots and roots

```{r}
# filtering the data by tissue to prepare for differential expression  
shoots <- expr_df |> filter(tissue == "Shoots") 
roots <- expr_df |> filter(tissue == "Roots")
```

## Differential expression

```{r}
# computing differential gene expression in tissue, taking into account conditions  
comp_de <- function(dat) {   
  dat |> 
    group_by(gene) |>  
    summarise(
      mean_cold = mean(expression[condition == "cold"], na.rm = TRUE),
      mean_control = mean(expression[condition == "control"], na.rm = TRUE),
      log2FC = mean_cold - mean_control,
      pvalue = t.test(expression ~ condition)$p.value) |>   
    mutate(neglog10p = -log10(pvalue))}   

shoots_de <- comp_de(shoots) 
roots_de <- comp_de(roots)
```

## Trial volcano plot for gene expression in shoots

```{r}
# trying out the data on in the plot
ggplot(data = shoots_de, 
       aes(x = log2FC, y = neglog10p)) +   
  geom_vline(
    xintercept = c(-1, 1), 
    col = "gray", 
    linetype = "dashed") +   
  geom_hline(
    yintercept = -log10(0.05), 
    col = "gray", 
    linetype = "dashed") +    
  geom_point(alpha = 0.7) +   
  labs(     
    title = "Volcano plot - gene expression in shoots in control vs cold",     
    x = expression("log"[2]*" fold change"), 
    y = expression("-log"[10]*"p-value")   
    )
```

There is too many data points, so the plot is not optimal. To adjust that, the expression will be marked as up/down-regulated or not-significant.

```{r}
# marking significance of gene expression changes
res_shoots <- shoots_de |>    
  mutate(     
    sig = case_when(       
    pvalue < 0.05 & log2FC > 1  ~ "Upregulated",       
    pvalue < 0.05 & log2FC < -1 ~ "Downregulated",       
    TRUE ~ "Not significant") 
    )
```

## Updated volcano plot for gene expression in shoots

```{r}
# calculating gene expression means to get the general regulation of genes
gene_means <- expr_df |>     
  group_by(gene) |>    
  summarise(mean_expr = mean(expression))  

# adding the means to the data and capping the values that are above 10 to make have less outliers
res_shoots_clean <- res_shoots |>    
  left_join(gene_means, by="gene") |>
  mutate(
    neglog10p_capped = pmin(neglog10p, 10)
    ) |>    
  filter(mean_expr > 1)   

# adding labels to top 5 upregulated genes
top_up <- res_shoots_clean |>
  arrange(desc(log2FC)) |>
  slice_head(n = 5)

# adding labels to top 5 downregulated genes
top_down <- res_shoots_clean |>
  arrange(log2FC) |>
  slice_head(n = 5)

# pooling the labels to make it easier to add to the plot
top_labels <- bind_rows(top_up, top_down)

shoots_plot <- ggplot(res_shoots_clean, 
       aes(log2FC, neglog10p_capped)) +   
  geom_point(
    aes(color = sig,
        alpha = sig,
        size = sig)) +
  geom_text_repel(data = top_labels,   # using ggrepel to avoiding label collision
  aes(label = gene),
  box.padding = 0.5,
  point.padding = 0.3,
  segment.color = "grey40",
  segment.size = 0.5,
  arrow = arrow(length = unit(0.01, "npc")), 
  min.segment.length = 0.1, 
  max.overlaps = 20,
  size = 3,
  force = 2
  ) +
  scale_alpha_manual(values = c("Not significant" = 0.15,
                                "Upregulated" = 0.8,
                                "Downregulated" = 0.8)) + 
  scale_size_manual(values = c("Not significant" = 0.5,
                                "Upregulated" = 1.5,
                                "Downregulated" = 1.5)) + 
  scale_color_manual(values = c(
    "Upregulated"="#A30000", 
    "Downregulated"="#0000A3",
    "Not significant"="gray")) +   
  geom_vline(
    xintercept = c(-1, 1), 
    col = "gray", 
    linetype = "dashed") +   
  scale_x_continuous(
    breaks = seq(-3, 5, 1)) +   
  theme_light() +   
  theme(legend.position = "bottom") +   
  guides(alpha = "none", size = "none") + 
  labs(
    title = "Volcano plot - gene expression in shoots in control vs cold",
    color = "Expression rate", 
    x = expression("log"[2]*" fold change"), 
    y = expression("-log"[10]*"p-value"))

shoots_plot
```

## Volcano plot for gene expression in roots

```{r}
# marking significance of gene expression changes
res_roots <- roots_de |>    
  mutate(     
    sig = case_when(       
    pvalue < 0.05 & log2FC > 1  ~ "Upregulated",       
    pvalue < 0.05 & log2FC < -1 ~ "Downregulated",       
    TRUE ~ "Not significant") 
    )
```

```{r}
# calculating gene expression means to get the general regulation of genes
gene_means <- expr_df |>     
  group_by(gene) |>    
  summarise(mean_expr = mean(expression))  

# adding the means to the data and capping the values that are above 10 to make have less outliers
res_roots_clean <- res_roots |>    
  left_join(gene_means, by="gene") |>
  mutate(
    neglog10p_capped = pmin(neglog10p, 10)
    ) |>    
  filter(mean_expr > 1)   

# adding labels to top 5 upregulated genes
top_up <- res_roots_clean |>
  arrange(desc(log2FC)) |>
  slice_head(n = 5)

# adding labels to top 5 downregulated genes
top_down <- res_roots_clean |>
  arrange(log2FC) |>
  slice_head(n = 5)

# pooling the labels to make it easier to add to the plot
top_labels <- bind_rows(top_up, top_down)

roots_plot <- ggplot(res_roots_clean, 
       aes(log2FC, neglog10p_capped)) +   
  geom_point(
    aes(color = sig,
        alpha = sig,
        size = sig)) +
  geom_text_repel(data = top_labels,  # using ggrepel to avoiding label collision
  aes(label = gene),
  box.padding = 0.5,
  point.padding = 0.3,
  segment.color = "grey40",
  segment.size = 0.5,
  arrow = arrow(length = unit(0.01, "npc")),
  min.segment.length = 0.1,  
  max.overlaps = 20,
  size = 3,
  force = 2
  ) +
  scale_alpha_manual(values = c("Not significant" = 0.15,
                                "Upregulated" = 0.8,
                                "Downregulated" = 0.8)) + 
  scale_size_manual(values = c("Not significant" = 0.5,
                                "Upregulated" = 1.5,
                                "Downregulated" = 1.5)) + 
  scale_color_manual(values = c(
    "Upregulated"="#A30000", 
    "Downregulated"="#0000A3",
    "Not significant"="gray")) +   
  geom_vline(
    xintercept = c(-1, 1), 
    col = "gray", 
    linetype = "dashed") +   
  scale_x_continuous(
    breaks = seq(-3, 5, 1)) +   
  theme_light() +   
  theme(legend.position = "bottom") +   
  guides(alpha = "none", size = "none") + 
  labs(
    title = "Volcano plot - gene expression in roots in control vs cold",
    color = "Expression rate", 
    x = expression("log"[2]*" fold change"), 
    y = expression("-log"[10]*"p-value"))

roots_plot
```
